### PR TITLE
[16.0][MIG] Migration of module mail_tracking_mailgun

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

Migration of module mail_tracking_mailgun

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### Notes

This PR depends on https://github.com/OCA/social/pull/1029

I have tested functionally with PR https://github.com/OCA/social/pull/1029 and cherry-pick https://github.com/OCA/social/pull/1056

### CODE REVIEW

File: mail_tracking_mailgun/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    @route(["/mail/tracking/mailgun/all"], auth="none", type="json", csrf=False)
+    def mail_tracking_mailgun_webhook(self):
+        """Process webhooks from Mailgun."""
+        ensure_db()
+        # Verify and return 406 in case of failure, to avoid retries
+        # See https://documentation.mailgun.com/en/latest/user_manual.html#routes
+        try:
+            self._mail_tracking_mailgun_webhook_verify(
+                **request.jsonrequest["signature"]
```
Comment (maq-adhoc) [inline mail_tracking_mailgun/controllers/main.py]: Hi, The request object is not the same in version 16 .
`                **request.jsonrequest["signature"]
`
should be  
`                 **request.dispatcher.jsonrequest["signature"]`

the same for event-data

Thanks

Comment (maq-adhoc) [reply mail_tracking_mailgun/controllers/main.py]: @alan196 hi could you see this?

**Comment by chienandalu** [comment]
I don't know if this PR is in progress. In any case, take this recent patch into consideration: https://github.com/OCA/social/pull/1190

**Comment by tobru** [comment]
I gave this PR a try in an Odoo 16 test environment and so far, it looks good, works as expected.

**Pull request closed without merging**